### PR TITLE
Add Resource modifiers docs / site

### PR DESCRIPTION
**Pull request by anshulahuja98**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Docs for : #6452 
# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: site/content/docs/main/restore-resource-modifiers.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+### Operations supported by the JSON Patch RFC: 
+- add
+- remove
+- replace
+- move
+- copy
+- test (covered below)
+
+### Advance scenarios
```
Comment (sseago) [inline site/content/docs/main/restore-resource-modifiers.md:57]: s/Advance/Advanced/

File: site/content/docs/main/restore-resource-modifiers.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+Below is the two-step of using resource modifiers to modify the resources during restore.
+1. Creating resource modifiers configmap
+
+   You need to create one configmap in Velero install namespace from a YAML file that defined resource modifiers. The creating command would be like the below:
+   ```bash
+   kubectl create cm <configmap-name> --from-file <yaml-file> -n velero
+   ```
+2. Creating a restore reference to the defined resource policies
+
+   You can create a restore with the flag `--resource-modifier-configmap`, which will reference the current restore to the defined resource modifiers. The creating command would be like the below:
```
Comment (sseago) [inline site/content/docs/main/restore-resource-modifiers.md:20]: I would replace "which will reference the current restore to the defined resource modifiers" with "which will apply the defined resource modifiers to the current restore".

**Review by sseago — COMMENTED** [review]
A couple minor suggestions on wording, otherwise looks good.

File: site/content/docs/main/restore-resource-modifiers.md (hunk 2 above)
Comment (anshulahuja98) [reply site/content/docs/main/restore-resource-modifiers.md]: Done.

File: site/content/docs/main/restore-resource-modifiers.md (hunk 1 above)
Comment (anshulahuja98) [reply site/content/docs/main/restore-resource-modifiers.md]: Done.

**Comment by anshulahuja98** [comment]
> A couple minor suggestions on wording, otherwise looks good.

Resolved the comments

**Review by sseago — APPROVED** [review]

**Comment by anshulahuja98** [comment]
@blackpiglet  / @ywk253100 can you please review.

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**